commit f4515855b7a1c2edf674e0f35b0f22abc9ed2165
Author: Selvin Xavier <selvin.xavier@broadcom.com>
Date:   Wed Nov 21 00:05:00 2018 -0800

    RDMA/bnxt_re: Fix system hang when registration with L2 driver fails
    
    [ Upstream commit 3c4b1419c33c2417836a63f8126834ee36968321 ]
    
    Driver doesn't release rtnl lock if registration with
    L2 driver (bnxt_re_register_netdev) fais and this causes
    hang while requesting for the next lock.
    
    [  371.635416] "echo 0 > /proc/sys/kernel/hung_task_timeout_secs" disables this message.
    [  371.635417] kworker/u48:1   D    0   634      2 0x80000000
    [  371.635423] Workqueue: bnxt_re bnxt_re_task [bnxt_re]
    [  371.635424] Call Trace:
    [  371.635426]  ? __schedule+0x36b/0xbd0
    [  371.635429]  schedule+0x39/0x90
    [  371.635430]  schedule_preempt_disabled+0x11/0x20
    [  371.635431]  __mutex_lock+0x45b/0x9c0
    [  371.635433]  ? __mutex_lock+0x16d/0x9c0
    [  371.635435]  ? bnxt_re_ib_reg+0x2b/0xb30 [bnxt_re]
    [  371.635438]  ? wake_up_klogd+0x37/0x40
    [  371.635442]  bnxt_re_ib_reg+0x2b/0xb30 [bnxt_re]
    [  371.635447]  bnxt_re_task+0xfd/0x180 [bnxt_re]
    [  371.635449]  process_one_work+0x216/0x5b0
    [  371.635450]  ? process_one_work+0x189/0x5b0
    [  371.635453]  worker_thread+0x4e/0x3d0
    [  371.635455]  kthread+0x10e/0x140
    [  371.635456]  ? process_one_work+0x5b0/0x5b0
    [  371.635458]  ? kthread_stop+0x220/0x220
    [  371.635460]  ret_from_fork+0x3a/0x50
    [  371.635477] INFO: task NetworkManager:1228 blocked for more than 120 seconds.
    [  371.635478]       Tainted: G    B      OE     4.20.0-rc1+ #42
    [  371.635479] "echo 0 > /proc/sys/kernel/hung_task_timeout_secs" disables this message.
    
    Release the rtnl_lock correctly in the failure path.
    
    Fixes: de5c95d0f518 ("RDMA/bnxt_re: Fix system crash during RDMA resource initialization")
    Signed-off-by: Selvin Xavier <selvin.xavier@broadcom.com>
    Signed-off-by: Jason Gunthorpe <jgg@mellanox.com>
    Signed-off-by: Sasha Levin <sashal@kernel.org>

diff --git a/drivers/infiniband/hw/bnxt_re/main.c b/drivers/infiniband/hw/bnxt_re/main.c
index 85cd1a3593d6..bb3f16273938 100644
--- a/drivers/infiniband/hw/bnxt_re/main.c
+++ b/drivers/infiniband/hw/bnxt_re/main.c
@@ -1252,6 +1252,7 @@ static int bnxt_re_ib_reg(struct bnxt_re_dev *rdev)
 	/* Registered a new RoCE device instance to netdev */
 	rc = bnxt_re_register_netdev(rdev);
 	if (rc) {
+		rtnl_unlock();
 		pr_err("Failed to register with netedev: %#x\n", rc);
 		return -EINVAL;
 	}
diff -uN a/1.txt b/1.txt
--- a/dummy/rpi_1072_f4515855b7a1c2edf674e0f35b0f22abc9ed2165.txt	1970-01-01 00:00:00.000000000 +0000
+++ b/dummy/rpi_1072_f4515855b7a1c2edf674e0f35b0f22abc9ed2165.txt	2013-12-23 04:07:40.000000000 +0000
@@ -0,0 +1 @@
+dummy file to ensure patch has content.
